client/landing/stepper/declarative-flow/internals/steps-repository/domain-transfer-domains/domains.tsx

```diff
@@ -261,9 +261,11 @@ const Domains: React.FC< Props > = ( { onSubmit, variantSlug } ) => {
 					variantSlug={ variantSlug }
 				/>
 			) ) }
-			<Button className="bulk-domain-transfer__add-domain" icon={ plus } onClick={ addDomain }>
-				{ isHundredYearTransfer ? __( 'Add another domain' ) : __( 'Add more' ) }
-			</Button>
+			{ ! isHundredYearTransfer && (
+				<Button className="bulk-domain-transfer__add-domain" icon={ plus } onClick={ addDomain }>
+					{ __( 'Add more' ) }
+				</Button>
+			) }
```

Comment:
Skipping the `Add more` button entirely since no more than one product is allowed for 100-year product purchases, as per validation in the backend, wouldn't make sense to have the button here. Plus, adding two domain transfers or more would leave only the last one in the cart.